firmware/src/msp.c

```diff
@@ -45,10 +46,42 @@ void send_msp_displayport_clear() {
   send_msp(182, payload, 1);
 }
 
+// Parts of this are borrowed from the QuickSilver firmware's source code
+uint8_t guess_battery_cell_count(uint8_t chemistry) {
+  // Get an average mV reading
+  int count = 0;
+  int vbatt_avg = 0;
+  while(count < 5000) {
```

Comment:
This might appear to be a good way to get an average, but I suspect you will just read the same ADC value 5,000 times. The CPU is much faster than the ADC. If you want an average, it would be better to program the ADC do do this in hardware.

Reply:
Teach me :)

Reply:
Thank you ... change pushed! :D